cpmonitor/admin.py

```diff
@@ -99,6 +99,34 @@ def mk_dropdown_tree(self, model, for_node=None):
             self.__class__.add_subtree(for_node, node, options)
         return options
 
+    def clean(self):
+        """Prepare uniqueness validation of `slugs` in case position or title changed in form.
+
+        This only updates `slugs` according to new parents `slugs` and title.
+        The actual validation is done by `validate_constraints` in the model.
+        The descendant's `slugs` are corrected during `instance.save()`.
+        """
+
+        # Copied from treebeard.forms.MoveNodeForm._clean_cleaned_data():
+        reference_node_id = None
+        if "_ref_node_id" in self.cleaned_data:
+            if self.cleaned_data["_ref_node_id"] != "0":
+                reference_node_id = self.cleaned_data["_ref_node_id"]
+                if reference_node_id.isdigit():
+                    reference_node_id = int(reference_node_id)
+        reference_node = None
+        if reference_node_id:
+            reference_node = self._meta.model.objects.get(pk=reference_node_id)
+        position_type = self.cleaned_data["_position"]
```

Comment:
This is slightly different from TreeBeard's impl (at least on their master branch). I suppose that was on purpose..?

Let's either use the same code and call it "copied", or adjust it if really needed and call it "based on", just for clarity...
```suggestion
        # Based on treebeard.forms.MoveNodeForm._clean_cleaned_data():
        reference_node_id = None
        if "_ref_node_id" in self.cleaned_data:
            if self.cleaned_data["_ref_node_id"] != "0":
                reference_node_id = self.cleaned_data["_ref_node_id"]
                if reference_node_id.isdigit():
                    reference_node_id = int(reference_node_id)
        reference_node = None
        if reference_node_id:
            reference_node = self._meta.model.objects.get(pk=reference_node_id)
        position_type = self.cleaned_data["_position"]
```